### PR TITLE
[BPF] skip FIB when dest is unknown at HEP

### PR DESCRIPTION
**Pull request by tomastigera**
## Description

    [BPF] skip FIB when dest is unknown at HEP
    
    * fix when CALI_ST_SKIP_FIB is set on the way to the host, set
      CALI_CT_FLAG_SKIP_FIB on conntrack - not just when from WEP
    
    * add test for ^^^ and issue https://github.com/projectcalico/calico/issues/6450
    
    * In addition to skipping FIB when there is no route to post-dnat
      destination, also skip FIB when there is a route, but it is not local
      while there was no service involved. In that case, we are not
      forwarding a service (NodePort) to another node and we should only
      forward locally. Let the host decide what to do with such a packet.

    [BPF/UT] set RPF to strict as test provision for that.
    
    Make tests execute the path again after we introduced the option to
    disable RPF.

Fixes https://github.com/projectcalico/calico/issues/8918

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: fix - let the node handle packet when we are not sure about the destination
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Review by fasaxc — APPROVED** [review]
LGTM

**Pull request merged**
